src/block/icon-list/edit.js

```diff
@@ -243,12 +239,12 @@ const Edit = props => {
 
 		// Different icon sizes per device preview.
 		const { __experimentalGetPreviewDeviceType: getPreviewDeviceType } = select( 'core/edit-post' )
-		const previewDeviceType = getPreviewDeviceType()
+		const previewDeviceType = getPreviewDeviceType ? getPreviewDeviceType() : 'Desktop'
 		const currentIconSize = (
 			previewDeviceType === 'Desktop' ? props.attributes.iconSize :
 				previewDeviceType === 'Tablet' ? ( props.attributes.iconTabletSize || props.attributes.iconSize ) :
-					( props.attributes.iconMobileSize || props.attributes.iconTabletSize || props.attributes.iconSize )
-		) || 20
+					( props.attributes.iconMobileSize || props.attributes.iconTabletSize || props.attributes.iconSize ) ) ||
+		 20
```

Comment:
Please move all the WP 5.4 `getPreviewDeviceType` compatibility to another PR so we can have the global settings PR only about global settings